### ISSUE TITLE
Secure session flow - handling error case

### ISSUE THREAD

**Issue by goutham23** [DMTF/libspdm#2333]
I'm currently implementing the SPDM responder for the secure session and planning to support only SPDM cmds( GET_MEASUREMENTS, HEARTBEAT, END_SESSION) as application message
  /*
    * Secured message : Encrypted Application message
    * 1)Decode secured message to Application data
    * 2)Application message to SPDM message
    */
    
    After step 1), have a pointer pointing to the application message, highlighted below in the figure
    
![image](https://github.com/DMTF/libspdm/assets/36578776/785e9a45-7294-4646-ac7a-615e10a89939)

    
    But for some reason mctp_msg_type is not SPDM or the size of the application data len is less than the size of the SPDM message header structure (refer below). So there is not point in decoding the app message and I have to return the error to requester.
   
   
   QUESTION:
   Should I encrypt the error message as the sesion_id is valid? or just drop the response, clean the session info and no need to send anything back to the requester?
    
    
    typedef struct
{
    UINT8 spdm_version;             /**<
                                    * [7:4] The major version of the SPDM spec
                                    * [3:0] The minor version of the SPDM spec
                                    */
    UINT8 request_response_code;    /**< The request message code or response code */
    UINT8 param1;                   /**< The first one-byte parameter */
    UINT8 param2;                   /**< The second one-byte parameter */
} spdm_message_header_struct;

**Comment by steven-bellock** [comment]
>Should I encrypt the error message as the sesion_id is valid? or just drop the response, clean the session info and no need to send anything back to the requester?

That depends on the Responder's / Platform's security policy. If the Responder has a fixed security policy then follow that. If it relies on the Platform's security policy then the Responder's behavior may need to be configurable.

**Comment by goutham23** [comment]
I did not understand.

There is no policy defined, I'm following the spec and expect the Requester to follow the same.
Per spec, how do we handle this?

**Comment by jyao1** [comment]
2023/09/04: If the responder does not know encrypted message type, what should be returned?

Will file DSP0276 issue to ask.

**Comment by goutham23** [comment]
Thanks for filing, should I wait for an update in this ticket or is there a separate issue I can follow to know the updates on this

**Comment by goutham23** [comment]
@jyao1 

In this case, we can build the SPDM error response with error code - "SPDM_ERROR_CODE_UNEXPECTED_REQUEST" and encode the response and send it back to requester thru secure session.

Is this valid error case handling?

**Comment by jyao1** [comment]
We have discussed in the SPDM-WG meeting and we believe it should be MCTP error - `ERROR_UNSUPPORTED_CMD`.

See https://www.dmtf.org/sites/default/files/standards/documents/DSP0236_1.3.1.pdf, Table 13 – MCTP control message completion codes
